src/app/settings/views/Configuration/Security/SessionTimeout/SessionTimeout.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from "react";
+
+import { Spinner, Select, Button, Icon } from "@canonical/react-components";
+import { formatDuration } from "date-fns";
+import { useDispatch, useSelector } from "react-redux";
+
+import FormikField from "app/base/components/FormikField";
+import FormikForm from "app/base/components/FormikForm";
+import { actions as configActions } from "app/store/config";
+import configSelectors from "app/store/config/selectors";
+import {
+  durationToSeconds,
+  durationToHours,
+  secondsToDuration,
+} from "app/utils/timeSpan";
+
+type SessionTimeoutFormValues = {
+  session_length: number;
+  time_unit: "hours" | "days";
+};
+
+export enum Labels {
+  Loading = "Loading...",
+  Edit = "Edit",
+  Expiration = "Session timeout expiration",
+  Save = "Save",
+  Cancel = "Cancel",
+  TimeUnit = "Time unit",
+  Days = "Day(s)",
+  Hours = "Hour(s)",
+  ConfigureSessionTimeout = "Configure Session Timeout",
+}
+
+const SessionTimeout = (): JSX.Element => {
+  const dispatch = useDispatch();
+  const configLoading = useSelector(configSelectors.loading);
+  const sessionLength = useSelector(configSelectors.sessionLength) || undefined;
+  const saved = useSelector(configSelectors.saved);
+  const saving = useSelector(configSelectors.saving);
+
+  const [isFormOpen, setFormOpen] = useState(false);
+
+  const timeUnitOptions = [
+    { label: Labels.Hours, value: "hours" },
+    { label: Labels.Days, value: "days" },
+  ];
+
+  if (configLoading) {
+    return <Spinner aria-label={Labels.Loading} text={Labels.Loading} />;
+  }
+
+  if (!isFormOpen) {
+    return (
+      <>
+        <span className="u-text--muted">
+          Session timeout expiration
+          <Button
+            className="is-small p-button__show-form"
+            onClick={() => setFormOpen(true)}
+          >
+            <Icon name="edit" />
+            Edit
+          </Button>
+        </span>
+        <p>{formatDuration(secondsToDuration(sessionLength))}</p>
+      </>
+    );
+  } else
+    return (
+      <>
+        <FormikForm<SessionTimeoutFormValues>
+          aria-label={Labels.ConfigureSessionTimeout}
+          buttonsAlign="right"
+          buttonsBordered
+          cleanup={configActions.cleanup}
+          initialValues={{
+            session_length: durationToHours({ seconds: sessionLength }),
```

Comment:
Do you have a followup task to update this or are you always going to show hours, even when the saved value was more than a day?

Reply:
Yeah so for the values I'm gonna keep it as hours, although the text display (outside of edit mode) will be combined hours and days. Reason being is if the user sets an hours value that isn't divisible by 24, especially if it's > 24